### ISSUE TITLE
Perf of classification models

### ISSUE THREAD

**Issue by NicolasHug** [pytorch/vision#7603]
Not an issue, just posting here as a ref for myself.

Times are in ms for `(1, 3, 224 ,224)` images (either CF or CL, reported median over 100 exp), flop info  comes from our meta-data. Benchmark code below.

| name               |       CF |        CL |   CF / CL |    flop |   CF rank |   flop rank |   rank diff |
|:-------------------|---------:|----------:|----------:|--------:|----------:|------------:|------------:|
| alexnet            |  0.89124 |  0.971624 |  0.917269 |   0.714 |         1 |          17 |          16 |
| convnext_base      | 13.7237  | 13.7346   |  0.999209 |  15.355 |        43 |          55 |          12 |
| convnext_large     | 13.853   | 13.8906   |  0.997296 |  34.361 |        44 |          73 |          29 |
| convnext_small     | 13.6465  | 13.6416   |  1.00036  |   8.684 |        41 |          47 |           6 |
| convnext_tiny      |  7.84392 |  7.86515  |  0.997302 |   4.456 |        18 |          36 |          18 |
| densenet121        | 19.8938  | 21.7912   |  0.912931 |   2.834 |        60 |          26 |          34 |
| densenet161        | 27.3097  | 29.0633   |  0.939664 |   7.728 |        64 |          42 |          22 |
| densenet169        | 28.3263  | 31.1972   |  0.907977 |   3.36  |        67 |          29 |          38 |
| densenet201        | 33.6823  | 37.0577   |  0.908915 |   4.291 |        71 |          33 |          38 |
| efficientnet_b0    | 12.5915  | 13.1545   |  0.957204 |   0.386 |        36 |          11 |          25 |
| efficientnet_b1    | 17.3741  | 18.3571   |  0.946449 |   0.687 |        55 |          16 |          39 |
| efficientnet_b2    | 17.0899  | 17.9185   |  0.953755 |   1.088 |        54 |          21 |          33 |
| efficientnet_b3    | 19.2678  | 20.5609   |  0.937108 |   1.827 |        58 |          25 |          33 |
| efficientnet_b4    | 23.7256  | 25.1539   |  0.943214 |   4.394 |        62 |          34 |          28 |
| efficientnet_b5    | 28.3812  | 30.1952   |  0.939923 |  10.266 |        68 |          49 |          19 |
| efficientnet_b6    | 33.0436  | 35.1799   |  0.939276 |  19.068 |        70 |          65 |           5 |
| efficientnet_b7    | 39.8641  | 42.8595   |  0.93011  |  37.746 |        75 |          74 |           1 |
| efficientnet_v2_l  | 50.748   | 54.0905   |  0.938205 |  56.08  |        76 |          75 |           1 |
| efficientnet_v2_m  | 37.0623  | 39.1236   |  0.947312 |  24.582 |        74 |          70 |           4 |
| efficientnet_v2_s  | 26.3253  | 28.1082   |  0.93657  |   8.366 |        63 |          45 |          18 |
| maxvit_t           | 32.0231  | 32.1036   |  0.997492 |   5.558 |        69 |          38 |          31 |
| mnasnet0_5         |  7.19691 |  7.72946  |  0.931102 |   0.104 |        15 |           3 |          12 |
| mnasnet0_75        |  7.86994 |  8.35537  |  0.941902 |   0.215 |        20 |           5 |          15 |
| mnasnet1_0         |  7.99962 |  8.48491  |  0.942805 |   0.314 |        22 |           9 |          13 |
| mnasnet1_3         |  7.98102 |  8.51006  |  0.937833 |   0.526 |        21 |          14 |           7 |
| mobilenet_v2       |  8.9802  |  9.54093  |  0.941229 |   0.301 |        25 |           8 |          17 |
| mobilenet_v3_large |  9.40669 |  9.66345  |  0.973429 |   0.217 |        29 |           6 |          23 |
| mobilenet_v3_small |  6.84604 |  7.16401  |  0.955615 |   0.057 |        14 |           2 |          12 |
| regnet_x_16gf      | 13.7088  | 13.0293   |  1.05215  |  15.941 |        42 |          62 |          20 |
| regnet_x_1_6gf     | 10.3845  | 10.9915   |  0.944773 |   1.603 |        32 |          22 |          10 |
| regnet_x_32gf      | 14.1083  | 15.3907   |  0.916676 |  31.736 |        45 |          71 |          26 |
| regnet_x_3_2gf     | 13.5509  | 14.7354   |  0.919617 |   3.177 |        39 |          28 |          11 |
| regnet_x_400mf     | 11.6309  | 12.7006   |  0.915778 |   0.414 |        34 |          13 |          21 |
| regnet_x_800mf     |  8.92109 |  9.78482  |  0.911727 |   0.8   |        24 |          18 |           6 |
| regnet_x_8gf       | 13.6388  | 13.3845   |  1.019    |   7.995 |        40 |          44 |           4 |
| regnet_y_128gf     | 53.5696  | 57.8659   |  0.925755 | 374.57  |        77 |          77 |           0 |
| regnet_y_16gf      | 15.917   | 15.1868   |  1.04808  |  15.912 |        48 |          61 |          13 |
| regnet_y_1_6gf     | 19.7392  | 21.5479   |  0.916062 |   1.612 |        59 |          23 |          36 |
| regnet_y_32gf      | 16.4438  | 17.7788   |  0.924908 |  32.28  |        53 |          72 |          19 |
| regnet_y_3_2gf     | 15.8682  | 16.7403   |  0.947908 |   3.176 |        47 |          27 |          20 |
| regnet_y_400mf     | 11.8846  | 13.0359   |  0.911684 |   0.402 |        35 |          12 |          23 |
| regnet_y_800mf     | 10.869   | 11.6993   |  0.929033 |   0.834 |        33 |          20 |          13 |
| regnet_y_8gf       | 13.1428  | 14.4112   |  0.911983 |   8.473 |        37 |          46 |           9 |
| resnet101          | 16.2228  | 17.7558   |  0.913659 |   7.801 |        52 |          43 |           9 |
| resnet152          | 23.5251  | 25.8813   |  0.908961 |  11.514 |        61 |          53 |           8 |
| resnet18           |  3.05006 |  3.54913  |  0.859382 |   1.814 |        12 |          24 |          12 |
| resnet34           |  5.22787 |  6.01189  |  0.869588 |   3.664 |        13 |          30 |          17 |
| resnet50           |  8.98749 |  9.80739  |  0.9164   |   4.089 |        26 |          31 |           5 |
| resnext101_32x8d   | 16.0169  | 17.8728   |  0.89616  |  16.414 |        49 |          63 |          14 |
| resnext101_64x4d   | 16.0859  | 17.8992   |  0.898697 |  15.46  |        50 |          58 |           8 |
| resnext50_32x4d    |  8.99187 |  9.89514  |  0.908715 |   4.23  |        27 |          32 |           5 |
| shufflenet_v2_x0_5 |  7.8462  |  7.93355  |  0.988989 |   0.04  |        19 |           1 |          18 |
| shufflenet_v2_x1_0 |  9.45057 |  9.55879  |  0.988678 |   0.145 |        30 |           4 |          26 |
| shufflenet_v2_x1_5 |  9.05592 |  9.23774  |  0.980318 |   0.296 |        28 |           7 |          21 |
| shufflenet_v2_x2_0 |  9.58699 |  9.5958   |  0.999082 |   0.583 |        31 |          15 |          16 |
| squeezenet1_0      |  2.91531 |  3.14649  |  0.926528 |   0.819 |        10 |          19 |           9 |
| squeezenet1_1      |  2.88305 |  3.00394  |  0.959759 |   0.349 |         9 |          10 |           1 |
| swin_b             | 27.8251  | 27.8336   |  0.999692 |  15.431 |        66 |          57 |           9 |
| swin_s             | 27.6968  | 27.6573   |  1.00143  |   8.741 |        65 |          48 |          17 |
| swin_t             | 14.7992  | 14.8164   |  0.998833 |   4.491 |        46 |          37 |           9 |
| swin_v2_b          | 35.1934  | 36.2611   |  0.970555 |  20.325 |        72 |          68 |           4 |
| swin_v2_s          | 35.4706  | 35.3212   |  1.00423  |  11.546 |        73 |          54 |          19 |
| swin_v2_t          | 18.7334  | 18.7567   |  0.99876  |   5.94  |        57 |          39 |          18 |
| vgg11              |  1.28297 |  1.4635   |  0.876644 |   7.609 |         2 |          40 |          38 |
| vgg11_bn           |  1.86589 |  2.07849  |  0.897714 |   7.609 |         5 |          40 |          35 |
| vgg13              |  1.43657 |  1.65574  |  0.86763  |  11.308 |         3 |          50 |          47 |
| vgg13_bn           |  2.1571  |  2.38833  |  0.903182 |  11.308 |         7 |          50 |          43 |
| vgg16              |  1.72813 |  1.96554  |  0.879213 |  15.47  |         4 |          59 |          55 |
| vgg16_bn           |  2.56407 |  2.85293  |  0.898749 |  15.47  |         8 |          59 |          51 |
| vgg19              |  2.01975 |  2.27334  |  0.888451 |  19.632 |         6 |          66 |          60 |
| vgg19_bn           |  2.97386 |  3.32836  |  0.893492 |  19.632 |        11 |          66 |          55 |
| vit_b_16           |  7.46397 |  7.54048  |  0.989853 |  17.564 |        16 |          64 |          48 |
| vit_b_32           |  7.61235 |  7.71922  |  0.986156 |   4.409 |        17 |          35 |          18 |
| vit_l_16           | 18.2285  | 18.2687   |  0.997801 |  61.555 |        56 |          76 |          20 |
| vit_l_32           | 13.455   | 13.6343   |  0.986852 |  15.378 |        38 |          56 |          18 |
| wide_resnet101_2   | 16.1448  | 17.8338   |  0.905292 |  22.753 |        51 |          69 |          18 |
| wide_resnet50_2    |  8.90721 |  9.80383  |  0.908544 |  11.398 |        23 |          52 |          29 |

<details>

```py
import torch
import pandas as pd
import torchvision
from time import perf_counter_ns
from torchvision.models import list_models, get_model, get_model_weights
import pickle

classification_models = list_models(module=torchvision.models)

from collections import defaultdict
res = defaultdict(lambda: defaultdict(list))

num_runs = 100

names = []
formats = []
times = []

EXCLUDE = ("googlenet", "inception_v3", "vit_h_14")
for name in classification_models:
    if name in EXCLUDE:
        continue
    print(name)
    m = get_model(name).cuda().train()
    pre_processing = get_model_weights(name)["DEFAULT"].transforms(antialias=True)
    for memory_format in (torch.channels_last, torch.contiguous_format):
        for _ in range(num_runs):
            img = torch.randint(0, 256, size=(1, 3, 512, 512), dtype=torch.uint8)
            img = pre_processing(img)

            img = img.contiguous(memory_format=memory_format)
            img = img.cuda()

            torch.cuda.synchronize()
            start = perf_counter_ns()
            m(img)
            torch.cuda.synchronize()
            stop = perf_counter_ns()

            names.append(name)
            formats.append(str(memory_format))
            times.append(stop - start)


with open("res", "wb") as f:
    pickle.dump((names, formats, times), f)
```

```py
#%%
import pickle
import pandas as pd
pd.set_option('display.max_rows', 500)
#%%
with open("/home/nicolashug/res", "rb") as f:
    names, formats, times = pickle.load(f)
df = pd.DataFrame([names, formats, times], )
df = df.T
df.columns = ("name", "format", "time")
df["format"] = df["format"].apply(lambda x: "CL" if "last" in x else "CF")
df["time"] /= 1e6
df

#%%
# print(df.groupby(["name", "format"]).describe(""))
table = df.pivot_table(values="time", index=("name", "format"), aggfunc="median").unstack()
table = table.droplevel(0, axis="columns")
table = table.sort_index()
table

# %%
table["CF / CL"] = table["CF"] / table["CL"]
table
# %%

# %%

# %%

# %%

import torchvision
from torchvision.models import list_models, get_model, get_model_weights

ops = []
classification_models = list_models(module=torchvision.models)
EXCLUDE = ("googlenet", "inception_v3", "vit_h_14")
for name in classification_models:
    if name in EXCLUDE:
        continue
    w = get_model_weights(name)["DEFAULT"]
    ops.append((name, w.meta["_ops"]))

ops = sorted(ops, key=lambda x: x[0])
ops = [v for (name , v) in ops]
# %%
table["flop"] = ops
table

# %%
table["CF rank"] = table["CF"].rank().astype(int)
table["flop rank"] = table["flop"].rank().astype(int)
table["rank diff"] = abs(table["CF rank"] - table["flop rank"])
table
# %%
print(table.to_markdown())
#

```

</details>
